src/Model/Widget/SpanParser.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpTui\Tui\Model\Widget;
+
+use PhpTui\Tui\Model\Color;
+use PhpTui\Tui\Model\Color\AnsiColor;
+use PhpTui\Tui\Model\Color\RgbColor;
+use PhpTui\Tui\Model\Modifier;
+use PhpTui\Tui\Model\Style;
+
+final class SpanParser
+{
+    private const OPEN_TAG_REGEX = '[a-z](?:[^\\\\<>]*+ | \\\\.)*';
+    private const CLOSE_TAG_REGEX = '[a-z][^<>]*+';
+
+    /** @var array<Style> */
+    private array $styleStack = [];
+
+    public static function new(): self
+    {
+        return new self();
+    }
+
+    /**
+     * @return array<Span>
+     */
+    public function parse(string $input): array
+    {
+        $regex = sprintf('#<((%s) | /(%s)?)>#ix', self::OPEN_TAG_REGEX, self::CLOSE_TAG_REGEX);
```

Comment:
in Symfony the delimiter is `;` not space...

Reply:
Yes, I didn't strictly follow them, tbh. Do you prefer `;`?

Reply:
yes, let's make it the same as Symfony

Reply:
Deal!